src/components/createContainer.jsx

```diff
@@ -0,0 +1,108 @@
+/* @flow */
+
+import React, { Component, PropTypes } from 'react';
+import invariant from 'invariant';
+
+import getDisplayName from '../utils/getDisplayName';
+import createAdaptorShape from '../utils/createAdaptorShape';
+
+const adaptorShape = createAdaptorShape(PropTypes);
+
+export default function createContainer(DecoratedComponent, specs) {
```

Comment:
First let me explain why I removed some of your things. Some of those was not very clear for me so I decided to start from introducing not all the things. So I've started with adaptor idea, uncommitted variables idea and multiple queries idea. We definitely would introduce all your significant changes in this PR with later commits to grow from simple things to more complex things.

Personally I think that `setArgs` and mutable args itself is rigid idea. I absolutely like your approach with more reactive style. I would like to totally remove `initialArgs` and `setArgs` so the only way would be to define `args` as a pure function of props. But first we need to think more about if is it possible to do all the things your want with only this approach.